Wrong error message when calling static method using dot notation

[Playpen](http://is.gd/iisPSu). For some reasons I wasn't able to get this error message when struct is in the same crate.

``` rust
extern crate rustc;

use rustc::middle::ty::MethodCall;

fn main() {
    let mc = MethodCall.expr(); // ERROR: unresolved name `MethodCall` [E0425]
    // let mc = MethodCall; // same error here
}
```

Here's the error message I get when struct is in the same crate

``` rust
<anon>:10:13: 10:16 error: `Foo` is the name of a struct or struct variant, but this expression uses it like a function name [E0423]
```

I think it should be raised in the case with crate too.

I believe the error message today indicates the problem fairly well, though it could show the `Foo::foo` form, so not closing.

```rust
struct Foo;

impl Foo {
    fn foo() {
    }
}

fn bar() {
    Foo.foo();
}
```

```
error[E0599]: no method named `foo` found for type `Foo` in the current scope
 --> test.rs:9:9
  |
9 |     Foo.foo();
  |         ^^^
  |
  = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
note: candidate #1 is defined in an impl for the type `Foo`
 --> test.rs:4:5
  |
4 | /     fn foo() {
5 | |     }
  | |_____^

error: aborting due to previous error(s)
```

--- a/src/librustc_typeck/check/method/suggest.rs
+++ b/src/librustc_typeck/check/method/suggest.rs
@@ -260,6 +260,7 @@ impl<'a, 'gcx, 'tcx> FnCtxt<'a, 'gcx, 'tcx> {
                 if !static_sources.is_empty() {
                     err.note("found the following associated functions; to be used as methods, \
                               functions must have a `self` parameter");
+                    err.help(&format!("try with `{}::{}`", self.ty_to_string(actual), item_name));
 
                     report_candidates(&mut err, static_sources);
                 }

--- a/src/test/ui/issue-41652/issue_41652.stderr
+++ b/src/test/ui/issue-41652/issue_41652.stderr
@@ -5,6 +5,7 @@ error[E0599]: no method named `f` found for type `{integer}` in the current scop
    |           ^
    |
    = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
+   = help: try with `{integer}::f`
 note: candidate #1 is defined in the trait `issue_41652_b::Tr`
   --> $DIR/auxiliary/issue_41652_b.rs:14:5
    |

--- a/src/test/ui/span/issue-7575.stderr
+++ b/src/test/ui/span/issue-7575.stderr
@@ -5,6 +5,7 @@ error[E0599]: no method named `f9` found for type `usize` in the current scope
    |                  ^^
    |
    = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
+   = help: try with `usize::f9`
 note: candidate #1 is defined in the trait `CtxtFn`
   --> $DIR/issue-7575.rs:16:5
    |
@@ -36,6 +37,7 @@ error[E0599]: no method named `fff` found for type `Myisize` in the current scop
    |                              ^^^
    |
    = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
+   = help: try with `Myisize::fff`
 note: candidate #1 is defined in an impl for the type `Myisize`
   --> $DIR/issue-7575.rs:51:5
    |
@@ -51,6 +53,7 @@ error[E0599]: no method named `is_str` found for type `T` in the current scope
    |       ^^^^^^
    |
    = note: found the following associated functions; to be used as methods, functions must have a `self` parameter
+   = help: try with `T::is_str`
 note: candidate #1 is defined in the trait `ManyImplTrait`
   --> $DIR/issue-7575.rs:57:5
    |

